airbyte-workers/src/main/java/io/airbyte/workers/process/KubePodProcess.java

```diff
@@ -504,10 +501,25 @@ public KubePodProcess(final boolean isOrchestrator,
         .endSpec()
         .build();
 
+    exitCodeFuture = new CompletableFuture<>();
+
     LOGGER.info("Creating pod...");
     val start = System.currentTimeMillis();
+
     this.podDefinition = fabricClient.pods().inNamespace(namespace).createOrReplace(pod);
 
+    // create the watch before the initialization runs
+    podWatch = fabricClient.resource(podDefinition).watch(new ExitCodeWatcher(
+        exitCodeFuture::complete,
+        exception -> {
```

Comment:
should we actually add the exception to the message here? in your experience, have you seen this for cases other than job cancellation?

would it be more accurate to say "Pod exit code watcher was shut down" instead of "Unable to find pod to retrieve exit value"?

Reply:
updated the wording

Reply:
I don't know that we can trust the exception doesn't include information about the pod manifest. It seems like we don't want to log that in case of failure?

Reply:
hmm, can we do something like e.getMessage to begin with? I agree with you - logging out just the message should reduce the risk of leaking anything. At the same time, it'll be nice to have as much information as possible. WDYT?

Reply:
sounds good